convert2rhel/pkghandler.py

```diff
@@ -72,29 +73,42 @@ def call_yum_cmd(command, args="", enablerepo=None, disablerepo=None,
     """Call yum command and optionally print its output."""
     loggerinst = logging.getLogger(__name__)
 
-    cmd = "yum %s -y" % (command)
+    cmd = "yum %s -y" % command
     # disablerepo parameter must be added before the enablerepo parameter
 
     if disablerepo is None:
         disablerepo = []
     if disablerepo:
         repos = disablerepo
-    else:
+    elif tool_opts.disablerepo:
         repos = tool_opts.disablerepo
+    else:
+        repos = ['*']
     for repo in repos:
-        cmd += " --disablerepo=%s " % repo
+        cmd += " --disablerepo=%s" % repo
 
     if enablerepo is None:
         enablerepo = []
     if enablerepo:
         repos = enablerepo
     else:
         repos = tool_opts.enablerepo
+
+    if not tool_opts.disable_submgr:
+        if not system_info.submgr_enabled_pkgs:  # double check to get all RHSM enabled repo if not get yet
+            subscription.get_enabled_repos()
+
+        for repo in system_info.submgr_enabled_pkgs:
+            repos.append(repo)
```

Comment:
Add tests to make sure that the repos from submgr is added to the command. As well as a test that makes sure that `--disablerepo` can disable repos from submgr

Reply:
These code https://github.com/oamg/convert2rhel/pull/84/files#diff-46f43c5ca8816e809009b34648b60673R150   test if a repo was added in the command, because if the repo is enabled, is was added by the function..

Reply:
That test is just for the `get_enabled_repos()` function itself, which is great. But we need tests for `call_yum_command()` that the repos from submgr is added to the resulting command and as well as a test that ensures RHSM repos isn't added to `--enablerepo` if the user passed it in `--disablerepo`

Reply:
@SpyTec already added on my lastest commit the unit test for `call_yum_command`
https://github.com/oamg/convert2rhel/pull/84/files#diff-cdec7561cf20f5b5b314293008180b42R83

Reply:
Ah, I see. So that one covers `elif tool_opts.disablerepo:` but it doesn't cover 
```py
if not tool_opts.disable_submgr:
    for repo in system_info.submgr_enabled_repos:
        repos.append(repo)
```
For that two tests should be created
1. if `disable_submgr` is true, `system_info.submgr_enabled_repos` should not be appended to repos var.
1. if `disable_submgr` is false, `system_info.submgr_enabled_repos` should be appended to repos var.

Then add a different test for when the user has `--disablerepo` for a RHSM repo e.g. `--disablerepo=rhel-7-server-rpms`, we should not add this repo to `--enablerepo`. Otherwise we'll end up with a command that looks like `yum install -y --enablerepo=rhel-7-server-rpms --disablerepo=rhel-7-server-rpms` which is rather ambiguous and should be avoided.

Reply:
I agree. The cases @SpyTec mentions in the previous comment is not covered yet.

Reply:
It's already created on my last commit, I believe :)